packages/ra-ui-materialui/src/list/datagrid/DatagridRow.spec.tsx

```diff
@@ -150,19 +167,27 @@ describe('<DatagridRow />', () => {
                 </RecordContextProvider>
             );
             expect(screen.queryAllByText('expanded')).toHaveLength(0);
-            fireEvent.click(screen.getByText('hello'));
+            const cell = screen.getByText('hello');
+            const row = cell.closest('tr');
+            if (!row) {
```

Comment:
Why not use `expect(row).toBeDefined();`

Reply:
Because this test is made to please typescript, who complains on the following line that row may be undefined when trying to access the row `classList`. It works with my test, not with `expect(row).toBeDefined();`